### ISSUE TITLE
Auto-saving/Auto save states

### ISSUE THREAD

**Issue by hellotinh03** [citra-emu/citra#6823]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

The auto-saving feature has been in many softwares, and Citra could make use of it as well.

Every 10 minutes (or anytime that the user choose), Citra will make a save state. Those save states will be in their own category, about 3 of them (or maybe more idk), separated from the manual save states.

### Why would this feature be useful?

Sometimes sh*t happens, maybe something crashes, or a blackout, or just simply dropping your controller on your keyboard and it accidentally hits the F6 key, throwing 3 hours of gameplay right out the window (speaking from experience). When those things happen, a recovery save state can actually save you from having to redo the same stuff you just done for 3 hours.

**Comment by Justin-ojay** [comment]
Yeah, just lost some pokemon because my game crashed before I could save
